ui/src/definitely-typed/filters/definitelyTypedFilters.tsx

```diff
@@ -2,80 +2,174 @@ import React, { useEffect, useState } from 'react';
+  runInfo: DefinitelyTypedRunInfo;
+  data: DefinitelyTypedTypeRun[];
+  filter(options: DefinitelyTypedFiltersOptions): void;
 }
 
 export interface DefinitelyTypedFiltersOptions {
-    filterIn: RegExp | null;
-    filterOut: RegExp | null;
-    isShowingSuccesses: boolean;
-    isShowingWarnings: boolean;
-    isShowingErrors: boolean;
+  filterIn: RegExp | null;
+  filterOut: RegExp | null;
+  isShowingSuccesses: boolean;
+  isShowingWarnings: boolean;
+  isShowingErrors: boolean;
 }
 
-export function DefinitelyTypedFilters(props: DefinitelyTypedFiltersProps): JSX.Element {
-    const [filterOut, setFilterOut] = useState('');
-    const [filterIn, setFilterIn] = useState('');
-    const [isShowing, setIsShowing] = useState('' as keyof DefinitelyTypedRunInfo);
-    const debouncedFilterOut = useDebounce(filterOut, 500);
-    const debouncedFilterIn = useDebounce(filterIn, 500);
-
-    useEffect(() => {
-        props.filter(buildOptions());
-    }, [debouncedFilterOut, debouncedFilterIn, isShowing]);
-
-    function buildOptions(): DefinitelyTypedFiltersOptions {
-        return {
-            filterIn: filterIn ? new RegExp(filterIn, 'i') : null,
-            filterOut: filterOut ? new RegExp(filterOut, 'i') : null,
-            isShowingErrors: isShowing === 'error',
-            isShowingWarnings: isShowing === 'warning',
-            isShowingSuccesses: isShowing === 'success'
-        };
-    }
+export function DefinitelyTypedFilters(
+  props: DefinitelyTypedFiltersProps
+): JSX.Element {
+  const [filterOut, setFilterOut] = useState('');
+  const [filterIn, setFilterIn] = useState('');
+  const [errorPresetFilterId, setErrorPresetFilterId] = useState(0);
+  const [isShowing, setIsShowing] = useState(
+    '' as keyof DefinitelyTypedRunInfo
+  );
+  const debouncedFilterOut = useDebounce(filterOut, 500);
+  const debouncedFilterIn = useDebounce(filterIn, 500);
+
+  useEffect(() => {
+    props.filter(buildOptions());
+  }, [debouncedFilterOut, debouncedFilterIn, isShowing, errorPresetFilterId]);
 
-    const infoKeys: Array<keyof DefinitelyTypedRunInfo> = ['total', 'success', 'warning', 'error'];
-    const infoLabelMap: {[key in keyof DefinitelyTypedRunInfo]: string} = {
-        error: 'Errors',
-        success: 'Passed',
-        warning: 'Warnings',
-        total: 'Total Types'
+  function buildOptions(): DefinitelyTypedFiltersOptions {
+    return {
+      filterIn: buildFilterInOption(),
+      filterOut: filterOut ? new RegExp(filterOut, 'i') : null,
+      isShowingErrors: isShowing === 'error',
+      isShowingWarnings: isShowing === 'warning',
+      isShowingSuccesses: isShowing === 'success',
     };
+  }
+
+  function buildFilterInOption() {
+    if (isShowing === 'error') {
+      return errorPresetFilterId === 1 && filterIn
+        ? new RegExp(filterIn, 'i')
+        : errorPresetFilterId === 0
+        ? null
+        : errorPresetFilterId !== 1
+        ? errorPresetFilters[errorPresetFilterId - 3].regex
```

Comment:
If you want to avoid the  -3 we could create the filter options first with all the values, then create the React Element from that list.

Reply:
addressed, with the enum the -3 is not anymore needed